BRDF w/test & Updated (passed in Material) ShadeRec w/test

kbarsalo opened the pull request:
Everything should work. Not super confident about the BRDF, if not correct then ignore this request and I'll fix it ASAP.

Somehow I broke something in the PointLight test, not sure how (see below for error details). I know there's been issues with team members not sharing broken tests, so it may just be on my end. 


       Expected 0 to equal 0.2672612419124244.
       Error: Expected 0 to equal 0.2672612419124244.
           at null.<anonymous> (c:/Users/Katherine/Documents/GitHub/tracejs/tes
s/unit/Lights/PointLightTest.js:56:35)
       Expected 0.7071067811865475 to equal 0.5345224838248488.
       Error: Expected 0.7071067811865475 to equal 0.5345224838248488.
           at null.<anonymous> (c:/Users/Katherine/Documents/GitHub/tracejs/tes
s/unit/Lights/PointLightTest.js:57:35)
       Expected 0.7071067811865475 to equal 0.8017837257372732.
       Error: Expected 0.7071067811865475 to equal 0.8017837257372732.
           at null.<anonymous> (c:/Users/Katherine/Documents/GitHub/tracejs/tes
s/unit/Lights/PointLightTest.js:58:35)

jdhaytko commented:
Hm, I assume this is in the test for get_direction()? For point light it
works by subtracting the hit point of shaderec from the location. Is there
any way that the method shaderec uses to calculate the hit point changed?

On Fri, Feb 13, 2015 at 12:24 AM, kbarsalo <notifications@github.com> wrote:

> Everything should work. Not super confident about the BRDF, if not correct
> then ignore this request and I'll fix it ASAP.
>
> Somehow I broke something in the PointLight test, not sure how (see below
> for error details). I know there's been issues with team members not
> sharing broken tests, so it may just be on my end.
>
>    Expected 0 to equal 0.2672612419124244.
>    Error: Expected 0 to equal 0.2672612419124244.
>        at null.<anonymous> (c:/Users/Katherine/Documents/GitHub/tracejs/tes
>
> s/unit/Lights/PointLightTest.js:56:35)
> Expected 0.7071067811865475 to equal 0.5345224838248488.
> Error: Expected 0.7071067811865475 to equal 0.5345224838248488.
> at null. (c:/Users/Katherine/Documents/GitHub/tracejs/tes
> s/unit/Lights/PointLightTest.js:57:35)
> Expected 0.7071067811865475 to equal 0.8017837257372732.
> Error: Expected 0.7071067811865475 to equal 0.8017837257372732.
> at null. (c:/Users/Katherine/Documents/GitHub/tracejs/tes
> s/unit/Lights/PointLightTest.js:58:35)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/camargo/tracejs/pull/26
> Commit Summary
>
>    - No complier errors for BRDF.ts
>    - Initial BRDF testing
>    - Material passed through ShadeRec fine
>
> File Changes
>
>    - *A* src/ts/Utilities/BRDF.ts
>    <https://github.com/camargo/tracejs/pull/26/files#diff-0> (33)
>    - *M* src/ts/Utilities/ShadeRec.ts
>    <https://github.com/camargo/tracejs/pull/26/files#diff-1> (5)
>    - *A* tests/unit/Utilities/BRDFTest.js
>    <https://github.com/camargo/tracejs/pull/26/files#diff-2> (17)
>    - *M* tests/unit/Utilities/ShadeRecTest.js
>    <https://github.com/camargo/tracejs/pull/26/files#diff-3> (8)
>
> Patch Links:
>
>    - https://github.com/camargo/tracejs/pull/26.patch
>    - https://github.com/camargo/tracejs/pull/26.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/camargo/tracejs/pull/26>.
>

camargo commented:
hmm master does not have this test error, have you changed anything in there? Also can you add BRDF.ts to a new folder called BRDFs and also add BRDFTest.js to a new folder in unit called BRDFs?

After adding these folders you will have to add a new task to the gulpfile.js: 

gulp.task('ts-to-js-brdfs', function() {
    return gulp.src('./src/ts/BRDFs/*.ts')
                               .pipe(ts(tsProject))
                               .pipe(gulp.dest('./src/js/BRDFs/'));

And then add 'ts-to-js-brdfs' to the default task.

kbarsalo commented:
Sorry just saw this message now.

John you're absolutely right, I added Material to be passed in ShadeRec so
I modified your test file. Works fine now.

I'll re-pull shortly.
